crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -1043,21 +1043,31 @@ impl<TX: DbTx> BlockReader for DatabaseProvider<TX> {
     /// will return None.
     fn block_with_senders(
         &self,
-        block_number: BlockNumber,
+        block_number: BlockHashOrNumber,
         transaction_kind: TransactionVariant,
     ) -> RethResult<Option<BlockWithSenders>> {
-        let Some(header) = self.header_by_number(block_number)? else { return Ok(None) };
+        let block_num: BlockNumber = match block_number {
+            BlockHashOrNumber::Hash(hash) => {
+                let res = match self.block_number(hash)? {
+                    Some(num) => Ok(num),
+                    None => Err("Block number not found for given hash"),
+                };
+                res.unwrap()
+            }
+            BlockHashOrNumber::Number(num) => num,
+        };
```

Comment:
can we make use of existing helper function here? https://github.com/paradigmxyz/reth/blob/4865e59684fbf1457125feec0c06a30405bf4b60/crates/storage/provider/src/traits/block_id.rs#L99-L105

@mattsse wdyt?

Reply:
sgtm!